usaspending_api/etl/award_helpers.py

```diff
@@ -142,112 +218,141 @@ def update_contract_awards(award_tuple=None):
-    #     for sa in sas:
-    #         a.total_subaward_amount += sa.amount
-    #     a.save()
-    # return rows
-
-    sql_sub_totals = (
-        "subaward_totals AS ("
-        "SELECT award_id, SUM(amount) AS total_subaward_amount, COUNT(*) AS subaward_count "
-        "FROM subaward "
+    _sql_sub_totals = str(
+        "subaward_totals AS ( "
+        "  SELECT "
+        "    award_id, "
+        "    SUM(amount) AS total_subaward_amount, "
+        "    COUNT(*) AS subaward_count "
+        "  FROM subaward "
+        "  {} "
+        "  GROUP BY award_id"
+        ") "
     )
     if award_tuple:
-        sql_sub_totals += "WHERE award_id IN %s "
-    sql_sub_totals += "GROUP BY award_id) "
+        values = [award_tuple]
+        sql_sub_totals = _sql_sub_totals.format("WHERE award_id IN %s ")
+    else:
+        values = None
+        sql_sub_totals = _sql_sub_totals.format("")
 
-    # Construct the SQL update
-    sql_update = "WITH {}".format(sql_sub_totals)
-    sql_update += (
+    _sql_update = str(
+        "WITH {} "
         "UPDATE awards "
-        "SET total_subaward_amount = subaward_totals.total_subaward_amount, "
-        "subaward_count = subaward_totals.subaward_count "
-        "FROM subaward_totals "
-        "WHERE subaward_totals.award_id = id"
+        "  SET "
+        "    total_subaward_amount = subaward_totals.total_subaward_amount, "
+        "    subaward_count = subaward_totals.subaward_count "
+        "  FROM subaward_totals "
+        "  WHERE subaward_totals.award_id = id "
     )
 
-    with connection.cursor() as cursor:
-        # If another expression is added and includes %s, you must add the tuple for that string interpolation to this
-        # list (even if it uses the same one!)
-        cursor.execute(sql_update, [award_tuple])
-        rows = cursor.rowcount
-
-    return rows
-
+    sql_update = _sql_update.format(sql_sub_totals)
+    return execute_database_insert_statement(sql_update, values)
+
+
+def update_award_categories(award_tuple: Optional[tuple] = None) -> int:
+    """This sets the category variable for an award."""
+
+    _sql_update = str(
+        "UPDATE awards SET "
+        "  category = CASE "
+        "    WHEN type IN ('A', 'B', 'C', 'D') THEN 'contract' "
+        "    WHEN type IN ('02', '03', '04', '05') THEN 'grant' "
+        "    WHEN type in ('06', '10') THEN 'direct payment' "
+        "    WHEN type in ('07', '08') THEN 'loans' "
+        "    WHEN type = '09' THEN 'insurance' "
+        "    WHEN type = '11' THEN 'other' "
+        "    WHEN type LIKE 'IDV%%' THEN 'idv' "
+        "    ELSE NULL END "
+        "  FROM awards "
+        "  {} "
+    )
 
-def update_award_categories(award_tuple=None):
-    """
-    This sets the category variable for an award.
-    """
-    awards = Award.objects.all()
     if award_tuple:
-        awards = awards.filter(id__in=list(award_tuple))
-    awards.update(
-        category=Case(
```

Comment:
The SQL version is used in `update_awards()`, so it replaced  the ORM version. This function could be removed if the legacy scripts which still use it are removed.